hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java

```diff
@@ -435,6 +444,22 @@ private long init(List<DatanodeStorageReport> reports) {
     return Math.max(overLoadedBytes, underLoadedBytes);
   }
 
+  private void sortOverUtilizedNodes() {
+    LOG.info("Sorting over-utilized nodes by capacity" +
+        " to bring down top used datanode capacity faster");
+
+    if (overUtilized instanceof List) {
+      List<Source> list = (List<Source>) overUtilized;
+      list.sort(
+          (Source source1, Source source2) ->
```

Comment:
Do we need to consider StorageType (which is associated with Source)? E.g., suppose a DN has 2 storage types, one of which is highly utilized and the other is just above average. Do we want to first schedule the movement for the highly-utilized storage type on this node?

Reply:
Good idea, we should use utilization for storage type instead of datanode utilization. Will fix